sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -321,135 +330,168 @@ static int register_device()
-        if (response.retry_after_seconds > 0)
-        {
-          printf("\tRetry-after: %u seconds.", response.retry_after_seconds);
-        }
-      }
+      printf("\tRetry-after: %u seconds.", response.retry_after_seconds);
     }
+  }
 
-    MQTTClient_freeMessage(&message);
-    MQTTClient_free(topic);
+  return get_operation_status_free(MQTTCLIENT_SUCCESS, topic, message);
+}
+
+static az_result parse_operation_message(
+    char* topic,
+    int topic_len,
+    MQTTClient_message* message,
+    az_iot_provisioning_client_register_response* response,
+    az_iot_provisioning_client_operation_status* operation_status)
+{
+  int rc;
+
+  printf("Topic: %s\n", topic);
+
+  if (topic_len == 0)
+  {
+    // The length of the topic if there are one or more NULL characters embedded in topic,
+    // otherwise topic_len is 0.
+    topic_len = (int)strlen(topic);
+  }
+
+  az_span topic_span = az_span_init((uint8_t*)topic, topic_len);
+  az_span message_span = az_span_init((uint8_t*)message->payload, message->payloadlen);
+
+  // Parse the incoming message and payload
+  if (az_failed(
+          rc = az_iot_provisioning_client_parse_received_topic_and_payload(
+              &provisioning_client, topic_span, message_span, response)))
+  {
+    printf("Message from unknown topic, az_result return code %0x4\n", rc);
+    return rc;
+  }
+  printf("Received payload:\n");
+  print_az_span(message_span);
+  printf("Response status is %d.\n", response->status);
+
+  // Parse the operation status from a string to an enum
+  if (az_failed(rc = az_iot_provisioning_client_parse_operation_status(response, operation_status)))
+  {
+    printf("Failed to parse operation_status, az_result return code %0x4\n", rc);
+    return rc;
+  }
+
+  return AZ_OK;
+}
+
+static int send_operation_query_message(az_iot_provisioning_client_register_response response)
+{
+  int rc;
+
+  printf("sending operation query message\n");
+
+  // Get the topic to send the query message
+  if (az_failed(
+          rc = az_iot_provisioning_client_query_status_get_publish_topic(
+              &provisioning_client, &response, query_topic, sizeof(query_topic), NULL)))
+  {
+    printf("Unable to get query status publish topic, az_result return code %04x\n", rc);
+    return rc;
+  }
+
+  // IMPORTANT: Wait the recommended retry-after number of seconds before query
+  printf("Querying after %u seconds...\n", response.retry_after_seconds);
+  sleep_for_seconds(response.retry_after_seconds);
+
+  // Publish the query message
+  if ((rc = MQTTClient_publish(mqtt_client, query_topic, 0, NULL, 0, 0, NULL))
+      != MQTTCLIENT_SUCCESS)
+  {
+    printf("Failed to publish query status request, MQTTClient return code %d\n", rc);
+    return rc;
   }
 
   return MQTTCLIENT_SUCCESS;
 }
 
+static int get_operation_status_free(int rc, char* topic, MQTTClient_message* message)
+{
+  if (message)
+    MQTTClient_freeMessage(&message);
```

Comment:
Please use braces even if it is single line.

Reply:
ok :)